### PR TITLE
python311Packages.nettigo-air-monitor: refactor

### PR DESCRIPTION
**Pull request by fabaff**
Diff: https://github.com/bieniu/nettigo-air-monitor/compare/refs/tags/2.2.0...2.2.1

Changelog: https://github.com/bieniu/nettigo-air-monitor/releases/tag/2.2.1

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 266425` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.nam</li>
    <li>python310Packages.nettigo-air-monitor</li>
    <li>python310Packages.nettigo-air-monitor.dist</li>
    <li>python311Packages.nettigo-air-monitor</li>
    <li>python311Packages.nettigo-air-monitor.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/nettigo-air-monitor/default.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 buildPythonPackage rec {
   pname = "nettigo-air-monitor";
-  version = "2.2.0";
+  version = "2.2.1";
   format = "setuptools";
 
   disabled = pythonOlder "3.8";
```
Comment (natsukium) [inline pkgs/development/python-modules/nettigo-air-monitor/default.nix:20]:
```suggestion
  disabled = pythonOlder "3.10";
```
https://github.com/bieniu/nettigo-air-monitor/blob/727b9f2f23b81de266534db360c94b49afde4134/setup.py#L24

File: pkgs/development/python-modules/nettigo-air-monitor/default.nix [hunk 2]
```diff
@@ -23,7 +23,7 @@ buildPythonPackage rec {
     owner = "bieniu";
     repo = pname;
     rev = "refs/tags/${version}";
-    hash = "sha256-K8EiDb6B18No9RNbw2a7U+FJQaXrrcFf0hgt40r6Igo=";
+    hash = "sha256-24O9Yl0+boxDtyPW4tBTsk2iDGGXf8ofkDHu8B+GxhE=";
   };
 
   propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/nettigo-air-monitor/default.nix:29]: I guess we don't need `orjson` anymore.
https://github.com/bieniu/nettigo-air-monitor/blob/727b9f2f23b81de266534db360c94b49afde4134/setup.py#L25

File: pkgs/development/python-modules/nettigo-air-monitor/default.nix [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 buildPythonPackage rec {
   pname = "nettigo-air-monitor";
-  version = "2.2.0";
+  version = "2.2.1";
   format = "setuptools";
```
Comment (mweinelt) [inline pkgs/development/python-modules/nettigo-air-monitor/default.nix:18]:
```suggestion
  pyproject = true;
```

And add `setuptools` to `nativeBuildInputs`. Running `python setup.py install` is deprecated. Every update should migrate packages away from setuptools CLI usage.

**Comment by mweinelt** [comment]
Uh, that commit is a mix of a lot of things now.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 266425` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.nam</li>
    <li>python310Packages.nettigo-air-monitor</li>
    <li>python310Packages.nettigo-air-monitor.dist</li>
    <li>python311Packages.nettigo-air-monitor</li>
    <li>python311Packages.nettigo-air-monitor.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
> Uh, that commit is a mix of a lot of things now.

Yupp, that rebase went wrong. Should be fixed now.

**Comment by mweinelt** [comment]
The master branch is already on 2.2.1, so you're only refactoring the package. Please update the commit message accordingly.

**Comment by fabaff** [comment]
@ofborg eval

**Pull request merged**